ports/rp2/boards/SIL_RP2040_SHIM/board.json

```
@@ -0,0 +1,20 @@
+{
+    "deploy": [
+        "deploy.md"
+    ],
+    "docs": "",
+    "features": [
+        "Breadboard Friendly",
+        "RGB LED",
+        "SPI Flash",
+        "Micro USB"
+    ],
```

Please try and use a subset of the features at [MicroPython Downloads](https://micropython.org/download/). Does this look close?

```suggestion
    "features": [
        "Dual-core",
        "RGB LED",
        "External Flash",
        "USB"
    ],
```

---

Hmm, that's a tough one.  According to its intended use with a PoE-FeatherWing, it should probably have "Ethernet" and "PoE" listed.  That's the whole reason for the existence of this board: to use it with a PoE-FeatherWing.  But Ethernet and PoE are not on the board by itself.

Also, it follows the Feather form factor but it isn't really a "Feather" as such, so listing that may or may not be correct.

Any thoughts on that?

---

I see the Portenta H7 and Teensy 4.1 have "Ethernet" listed despite not having an RJ-45 on board, so they need extra hardware.

---

For Teensy 4.1 it's just the RJ45 connector. Everything else like the PHY chip is on the board. Same for the Portenta H7, as far as I can tell.

---

They don't have magnetics.  So they need magnetics + RJ45 or an integrated jack.  Where do you draw the line?

I would argue that if a user is looking for MicroPython boards that support Ethernet, they should find the RP2040-Shim because that's specifically what it was created for.

---

>  Where do you draw the line?

The RP2040 has no Ethernet peripheral built-in. So per se the RP2040 Shim is a RP2040 with pin rows that match the feather family of boards. I understand that you designed it to be used with the feather Ethernet POE, but it could as well e.g. be used with the Feather Airlift adapter or other feather peripherals.

---

🤷‍♂️  I guess you're making my point here: it's arbitrary and different people have different ideas of where the line is drawn and *what exactly the purpose of these "feature" tags is*.

I'd argue that if saying the RP2040-Shim shouldn't be marked with the Ethernet feature, neither should the Portenta H7 and Teensy 4.1.  That they have MAC and PHY hardware is irrelevant to the majority of people that expect to be able to plug in an RJ45 cable if hardware is listed as "having Ethernet".  For many people that is the *only* relevant definition of "having Ethernet": being able to plug in their cable.

In the context of MicroPython, where the relevant driver needs to be compiled into the image, my main purpose for this custom image vs another RP2040 is that it has the W5500 Ethernet driver compiled into it, so it works seamlessly with the PoE-FeatherWing for which it was made.  As such, the thing the user is downloading when they select this tag, can easily be interpreted as "this image you're going to download has the necessary built-in support for Ethernet".

No point in further arguing, I understand your view, and I understand mine.  I'd like to see what some other people think and what the exact intent of the feature tags is if someone can clarify.  Is it purely about the hardware or is it related to software support included in the binary image?

---

Looks like no one else has any guidance on this.  I assume that if @mattytrentini thought any different he would have included "Ethernet" in his suggested list of tags so I'll just approve this as suggested.

---

Oh for crying out loud that dang commit format checker. I just signed off what was suggested.  I don't know how to fix this.